src/sql/base/browser/ui/chart/chart.component.ts

```diff
@@ -0,0 +1,154 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Component, Inject, forwardRef, ChangeDetectorRef } from '@angular/core';
+import * as chartjs from 'chart.js';
+import { mixin } from 'sql/base/common/objects';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { BubbleChartPoint, ChartOptions, ScatterChartPoint } from 'azdata';
+
+@Component({
+	selector: 'chart-component',
+	templateUrl: decodeURI(require.toUrl('./chart.component.html'))
+})
+export class Chart<T extends ChartOptions> extends Disposable {
+
+	public _labels: string[];
+	public _type: any;
+	public _data: number[] | BubbleChartPoint | ScatterChartPoint;
+	public _colors: string | string[];
+	public _label: string;
+	public _borderColor: string | string[];
+	public _chart: any;
+
+	public _options: any = {
+		events: ['click', 'keyup'],
+		responsive: true,
+		maintainAspectRatio: false
+	};
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _changeRef: ChangeDetectorRef
+	) {
+		chartjs.Chart.register(...chartjs.registerables);
+		super();
+	}
+
+	ngAfterViewInit(): void {
+
+	}
+
+	public set type(val: any) {
+		if (val === 'horizontalBar') {
+			this._type = 'bar';
+			this._options = mixin({}, mixin(this._options, { indexAxis: 'y' }));
+		}
+		else {
+			this._type = val;
+		}
+		this._changeRef.detectChanges();
+	}
+
+	public set data(val: any) {
```

Comment:
This doesn't clear old values first, so if I set `data = {labels: firstLabels, colors: firstColors}` then `data = {labels: secondLabels}`, the result would be `{labels: secondLabels, colors: firstColors}`, right?  Is that intentional?